src/host/snapshots.h

```diff
@@ -20,68 +21,83 @@ namespace asynchost
   static constexpr auto snapshot_idx_delimiter = "_";
   static constexpr auto snapshot_committed_suffix = "committed";
 
-  std::optional<std::pair<size_t, size_t>>
-  get_snapshot_evidence_idx_from_file_name(const std::string& file_name)
+  bool is_snapshot_file(const std::string& file_name)
   {
-    // Returns snapshot evidence and evidence commit proof indices
-    auto commit_pos =
-      file_name.find(fmt::format(".{}", snapshot_committed_suffix));
-    if (commit_pos == std::string::npos)
+    return nonstd::starts_with(file_name, snapshot_file_prefix);
+  }
+
+  bool is_snapshot_file_committed(const std::string& file_name)
+  {
+    return file_name.find(snapshot_committed_suffix) != std::string::npos;
+  }
+
+  std::optional<size_t> get_evidence_commit_idx_from_file_name(
+    const std::string& file_name)
+  {
+    // Only returns an evidence commit index for 1.x committed snapshots.
+    // 1.x committed snapshots file names are of the form:
+    // "snapshot_X_Y.committed_Z" while 2.x+ ones are of the form:
+    // "snapshot_X_Y.committed"
```

Comment:
I think this is worth unit testing in isolation, and this can also test the `stoi` overrun mentioned below (eg `snapshot_1_2.committed_3helloimjunk`)